runtime/parachains/src/disputes.rs

```diff
@@ -1038,74 +964,19 @@ impl<T: Config> Pallet<T> {
+		}
+
 		let is_local = <Included<T>>::contains_key(&set.session, &set.candidate_hash);
 		if !is_local {
-			let mut spam_slots: Vec<u32> =
-				SpamSlots::<T>::get(&set.session).unwrap_or_else(|| vec![0; n_validators]);
-			let mut spam_filter_struck = false;
-			for (validator_index, spam_slot_change) in summary.spam_slot_changes {
-				let spam_slot = spam_slots
-					.get_mut(validator_index.0 as usize)
-					.expect("index is in-bounds, as checked above; qed");
-
-				if let SpamSlotChange::Inc = spam_slot_change {
-					if *spam_slot >= max_spam_slots {
-						spam_filter_struck = true;
-
-						// Find the vote by this validator and filter it out.
-						let first_index_in_set = set
-							.statements
-							.iter()
-							.position(|(_statement, v_i, _signature)| &validator_index == v_i)
-							.expect(
-								"spam slots are only incremented when a new statement \
-									from a validator is included; qed",
-							);
-
-						// Note that there may be many votes by the validator in the statement
-						// set. There are not supposed to be, but the purpose of this function
-						// is to filter out invalid submissions, after all.
-						//
-						// This is fine - we only need to handle the first one, because all
-						// subsequent votes' indices have been added to the filter already
-						// by the duplicate checks above. It's only the first one which
-						// may not already have been filtered out.
-						filter.remove_index(first_index_in_set);
-
-						// Removing individual statments can cause the dispute to become onesided.
-						// Checking that (again) is done after the loop. Remove the bit indices.
-						summary.new_participants.set(validator_index.0 as _, false);
-					}
-
-					// It's also worth noting that the `DisputeStateImporter`
-					// which produces these spam slot updates only produces
-					// one spam slot update per validator because it rejects
-					// duplicate votes.
-					//
-					// So we don't need to worry about spam slots being
-					// updated incorrectly after receiving duplicates.
-					*spam_slot += 1;
-				} else {
-					*spam_slot = spam_slot.saturating_sub(1);
-				}
-			}
-
-			// We write the spam slots here because sequential calls to
-			// `filter_dispute_data` have a dependency on each other.
-			//
-			// For example, if a validator V occupies 1 spam slot and
-			// max is 2, then 2 sequential calls incrementing spam slot
-			// cannot be allowed.
-			//
-			// However, 3 sequential calls, where the first increments,
-			// the second decrements, and the third increments would be allowed.
-			SpamSlots::<T>::insert(&set.session, spam_slots);
-
 			// This is only relevant in cases where it's the first vote and the state
 			// would hence hold a onesided dispute. If a onesided dispute can never be
 			// started, by induction, we can never enter a state of a one sided dispute.
-			if spam_filter_struck && first_votes {
+			if first_votes {
```

Comment:
This if does not seem to do anything anymore? It looks like it is only counting votes, but then nothing is done with them. Am I missing something?

Reply:
No I think you don't. I missed the fact that the vote count is not used anywhere.
I've removed the whole if.